sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java

```diff
@@ -100,13 +120,53 @@ public synchronized void setMetrics(
       }
     }
 
-    if (totalFrames == 0 && slowFrames == 0 && frozenFrames == 0) {
+    return new FrameCounts(totalFrames, slowFrames, frozenFrames);
+  }
+
+  private static final class FrameCounts {
+    private final int totalFrames;
+    private final int slowFrames;
+    private final int frozenFrames;
+
+    private FrameCounts(final int totalFrames, final int slowFrames, final int frozenFrames) {
+      this.totalFrames = totalFrames;
+      this.slowFrames = slowFrames;
+      this.frozenFrames = frozenFrames;
+    }
+  }
+
+  @SuppressWarnings("NullAway")
+  public synchronized void setMetrics(
+      final @NotNull Activity activity, final @NotNull SentryId sentryId) {
```

Comment:
`l`: I think this would improve readability a bit. I had to jump to here https://github.com/getsentry/sentry-java/blob/f6b0dc7a17408741396774173c2637b717e3cbe1/sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java#L172 to understand which id it is. Same applies for `takeMetrics`.
```suggestion
      final @NotNull Activity activity, final @NotNull SentryId transactionId) {
```